### ISSUE TITLE
`avg(distinct)` support

### ISSUE THREAD

**Issue by WinkerDu** [apache/datafusion#2408]
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

DataFusion does not `avg(distinct ..)`:

```sql
> select avg(distinct a), avg(distinct b) from values(1, 2) t(a, b);
Execution error: avg(DISTINCT) aggregations are not available
```

Somewhat confusingly DataFusion does appear to support a single distinct `avg` 

```
> select avg(distinct a) from values(1, 2) t(a, b);
+-------------------+
| avg(DISTINCT t.a) |
+-------------------+
| 1.0               |
+-------------------+
1 row(s) fetched.
Elapsed 0.015 seconds.
```

But it does this by rewriting effectively rewriting the query `select avg(distinct a) ` to `select avg(a) from (select distinct a ...)` in the SingleDistinctToGroupBy rule

**Describe the solution you'd like**
Support multile distinct avg aggregates:
https://github.com/apache/datafusion/blob/2d303346c7d44a78d317e7b7b959cbd94edfb891/datafusion/functions-aggregate/src/average.rs#L116-L118

**Describe alternatives you've considered**
Basically we could follow the model of `COUNT(distinct ...)` aggregates:

In this case, there is a separate accumulator type that is implemented for count distinct

https://github.com/apache/datafusion/blob/5d08325165c1a7b32e5e35164919e83d46735e98/datafusion/functions-aggregate/src/count.rs#L234-L233

**Additional context**

**Comment by qazxcdswe123** [comment]
take

**Comment by Jefffrey** [comment]
@qazxcdswe123 I see you mentioned you were too busy to continue with the related PRs: https://github.com/apache/datafusion/pull/15413#issuecomment-2780650412

I plan to build upon your PRs to hopefully get this issue over the line, let me know if you have any concerns or if you're able to pick this back up again 👍
